syntax = "proto3";

option go_package = "github.com/nihilKnight/grpc-section/plcidenrt-v1";

package plcide;

import "google/protobuf/timestamp.proto";

service PLCRuntimeService {
    rpc ExportLogs(LogExportRequest) returns (stream LogEntry)

}


// ==================== Logs ====================
message LogEntry {
  google.protobuf.Timestamp timestamp = 1;
  LogLevel level = 2;
  string component = 3;
  string message = 4;

  enum LogLevel {
    DEBUG = 0;
    INFO = 1;
    WARNING = 2;
    ERROR = 3;
    UNKNOWN = 4;
  }
}

message LogExportRequest {
    string format = 1;  // 导出格式（"CSV"|"JSON"|"BINARY"）
    uint32 chunk_size = 2;  // 分块大小（字节）
}
